src/nuklear.h

```diff
@@ -5461,4 +5461,12 @@ template<typename T> struct nk_alignof{struct Big {T x; char c;}; enum {
 #define NK_ALIGNOF(t) ((char*)(&((struct {char c; t _h;}*)0)->_h) - (char*)0)
 #endif
 
+#ifdef NK_IMPLEMENTATION
+#define STB_RECT_PACK_IMPLEMENTATION
+#define STB_TRUETYPE_IMPLEMENTATION
+#endif
+
+#define STBTT_malloc(x,u) nk_malloc((nk_handle){0},0,x)
```

Comment:
Please change it to C89 (something like `#define STBTT_malloc(x,u) do{ nk_handle _ = {0}; nk_malloc( _, 0, x ) }while(0)`).

Reply:
Problem: the stb_* files have extensive "//" comments, which are disallowed by C89.

What's the rationale for not supporting C99? It's been 2 decades...

Reply:
>Problem: the stb_* files have extensive "//" comments, which are disallowed by C89.

Let's try adding a filter changing `//` to `/* ... */` (linewise is enough even if it'll look ugly) to the python script.

>What's the rationale for not supporting C99? It's been 2 decades...

Yep, it's long time ago, but the embedded world, which is one of the targets of this lib is even slower :cry:. On the other hand, we could make a strict schedule when we'll switch over to C99. What about a poll running for 4 months in this repo issue tracker (as pinned issue)?

Reply:
> What about a poll running for 4 months in this repo issue tracker (as pinned issue)?

Ahhhh yeah, now I remember the pain myself^^
Keeping the ANSI C description is definitely a fun argument to take a look at nuklear and is the reason I initially found nuklear.
Writing a big library such as nuklear to that standard is not trivial. That's quite the achievement in it's self.
However, leaving the C89 in favor of C99, just because **another** library needs comments replaced is not a good argument, especially in the light of @vurtun 's amazing work.
I too have come across this exact moment and have always transformed stb's libraries to C89 in the past, though I'm not sure I still have the regex script for that...
Lastly, sean berret has often stated, that he writes in C89 **except** for the comments, as far as I'm aware and from my experience with stb_image, stb_image_write and stb_dxt, all his libraries are fully C89 compliant, once the comments are replaced. (Since MSVC, which stb compiles with, still technically does not support anything beyond C89)

tl;dr:
-I'm ok with a poll in general, but considering changing to C99 because of the comments is horrible idea and quite the metaphorical fuck you to @vurtun 
-Changing comments is trivial and other graphics&openGL projects maintain the ANSI C compatibility, like https://github.com/recp/cglm

Reply:
@SairesArt yeah, I fully agree with everything you wrote - I meant to propose 2025 or later in the poll as the last C89 year for Nuklear (I suppose libs like GL will also be "modern enough" at that point to make the switch feel natural and not become an outsider :wink:). But maybe the poll is not necessary.

Btw. without @vurtun's decision for ANSI C, Nuklear would never become such a good library. Respect!